crates/bootstrap_srv/src/http.rs

```diff
@@ -168,23 +213,41 @@ fn tokio_thread(
 
                 let app = app.clone();
                 let shutdown_handle = shutdown_handle.clone();
-                if let Some(tls_config) = &config.tls_config {
+                if let Some(tls_config) = &server_config.tls_config {
                     let tls_config = tls_config
                         .create_tls_config()
                         .await
                         .expect("Failed to create TLS config");
 
+                    let acceptor = RustlsAcceptor::new(tls_config);
```

Comment:
May not need to be part of this pr, but I feel like we're going to need another custom acceptor befor the tls one here to support doing the ip deny/ratelimit checks before tls upgrade when *not* in use trusted ip mode.

Reply:
These run inside out. So the acceptor that's chained onto the RustlsAcceptor runs before the Rustls acceptor does. The implementation is "call the inner acceptor if set, then apply my behavior to the stream and return it". So I think this is behaving like you want?

Reply:
Or maybe I'm not following the trusted IP logic

Reply:
Okay, cool, I was not understanding that correctly. So then, `preflight_ip_check` will be called in the accept code, and the ip will not actually be checked if we are in use_trusted_ip mode. Then, in the `handle_upgraded` it will be.

Thanks! I'm happy with this :+1: